src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -1770,617 +1796,1418 @@ HRESULT CorProfiler::RunAutoInstrumentationLoader(const ComPtr<IMetaDataEmit2>&
 
-    //
-    // Define a new static method IsAlreadyLoaded on the new type that has a bool return type and takes no arguments;
-    //
-    mdMethodDef alreadyLoadedMethodToken;
-    BYTE        already_loaded_signature[] = {
-        IMAGE_CEE_CS_CALLCONV_DEFAULT,
-        0,
-        ELEMENT_TYPE_BOOLEAN,
-    };
-    hr = metadata_emit->DefineMethod(new_type_def, WStr("IsAlreadyLoaded"), mdStatic | mdPrivate,
-                                     already_loaded_signature, sizeof(already_loaded_signature), 0, 0,
-                                     &alreadyLoadedMethodToken);
-    if (FAILED(hr))
+        hr = metadata_import->FindMethod(system_app_domain_manager_token, WStr("CreateDomainHelper"),
+                                         system_app_domain_manager_create_domain_helper_signature.Head(),
+                                         system_app_domain_manager_create_domain_helper_signature.Size(),
+                                         &system_app_domain_manager_create_domain_helper_token);
+        if (FAILED(hr))
+        {
+            Logger::Warn("ModifyAppDomainCreate: FindMethod System.AppDomainManager::CreateDomainHelper failed");
+            return hr;
+        }
+
+        ILRewriter rewriter(this->info_, nullptr, module_id, system_app_domain_manager_create_domain_helper_token);
+        hr = rewriter.Import();
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("ModifyAppDomainCreate: ILRewriter.Import System.AppDomainManager::CreateDomainHelper failed");
+            return hr;
+        }
+
+        ILInstr* pFirstInstr = rewriter.GetILList()->m_pNext;
+        ILInstr* pNewInstr   = NULL;
+
+        pNewInstr           = rewriter.NewILInstr();
+        pNewInstr->m_opcode = CEE_LDARGA_S;
+        pNewInstr->m_Arg8   = 2;
+        rewriter.InsertBefore(pFirstInstr, pNewInstr);
+
+        pNewInstr           = rewriter.NewILInstr();
+        pNewInstr->m_opcode = CEE_CALL;
+        pNewInstr->m_Arg32  = patch_app_domain_setup_method;
+        rewriter.InsertBefore(pFirstInstr, pNewInstr);
+
+        hr = rewriter.Export();
+
+        if (FAILED(hr))
+        {
+            Logger::Warn("ModifyAppDomainCreate: ILRewriter.Export System.AppDomainManager::CreateDomainHelper failed");
+            return hr;
+        }
+
+        if (IsDumpILRewriteEnabled())
+        {
+            mdToken      token = 0;
+            TypeInfo     typeInfo{};
+            WSTRING      methodName = WStr("CreateDomainHelper");
+            FunctionInfo caller(token, methodName, typeInfo, MethodSignature(), FunctionMethodSignature());
+            Logger::Info(GetILCodes("*** ModifyAppDomainCreate: Modified Code: ", &rewriter, caller, metadata_import));
+        }
+    }
+
+    return S_OK;
+}
+
+// clang-format off
+// This method will generate new type __DDVoidMethodType__ in target module.
+// C# code for created class:
+// public static class __DDVoidMethodType__
```

Comment:
`__DDVoidMethodType__` should we fix "brading here" from DD to OTel? Is it ok in the same PR or new one - I'm not exactly sure about the scope (how big is it)?

Reply:
It is small, but let me do it in separate PR right after this will be merged (or before this PR if you think it is important to not add any new DD prefixed methods).